marimo/_plugins/ui/_impl/table.py

```diff
@@ -421,6 +422,21 @@ def __init__(
         self._lazy = _internal_lazy
         self._page_size = page_size
 
+        # HACK: this is a hack to get the name of the variable that was passed in
+        dataframe_name = '"<data>"'
```

Comment:
Have this issue with UI and state, but use a post execution hook in case there is reassignment etc...

This does work though, so